libdnf/base/goal.cpp

```diff
@@ -905,6 +936,62 @@ void Goal::Impl::add_up_down_distrosync_to_goal(
     }
 }
 
+GoalProblem Goal::Impl::add_group_install_to_goal(
+    base::Transaction & transaction,
+    const std::string & spec,
+    const transaction::TransactionItemReason reason,
+    GroupJobSettings & settings) {
+    auto & cfg_main = base->get_config();
+    bool strict = settings.resolve_strict(cfg_main);
+    comps::GroupQuery group_query(base);
+    comps::GroupQuery group_query_names(group_query);
+    group_query_names.filter_name(spec, sack::QueryCmp::IGLOB);
+    group_query.filter_groupid(spec, sack::QueryCmp::IGLOB);
+    group_query |= group_query_names;
```

Comment:
GroupJobSetting contains icase, group id and group name setting but it is not used. Probably it was intended to be used here.

Reply:
Yes, I've just added a commit that implements it.